momepy/preprocessing.py

```diff
@@ -740,3 +748,240 @@ def _get_extrapolated_line(coords, tolerance, point=False):
+    """Selecting only the lines that are touching but not covered by
+    the ``rab_plus``.
+    If more than one LineString is incoming to ``rab_plus``, COINS algortthm
+    is used to select the line to be extended further.
+    """
+    # Feels a bit combersome ... Ideally there would be a DISJOINT predicate
+    incoming = edges.sjoin(rab_plus , predicate = 'touches')
+    incoming.rename(columns ={'index_right':'index_rab_plus'}, inplace = True )
+    idx_drop =  incoming.sjoin(rab_plus, predicate = 'covered_by').index
+    incoming.drop(idx_drop, axis=0, inplace =True)
+
+    #figuring out which ends of incoming edges needs to be connected to the center_pt
+    incoming['first_pt'] = incoming.geometry.apply(lambda x : geometry.Point( x.coords[0]))
+    incoming['dist_fisrt_pt'] = incoming.center_pt.distance(incoming.first_pt)
+    incoming['last_pt'] = incoming.geometry.apply(lambda x : geometry.Point( x.coords[-1]))
+    incoming['dist_last_pt'] = incoming.center_pt.distance(incoming.last_pt)
+    lines = []
+    for i, row in incoming.iterrows() :
+        if row.dist_fisrt_pt < row.dist_last_pt :
+            lines.append(geometry.LineString([row.first_pt, row.center_pt]))
+        else :
+            lines.append(geometry.LineString([row.last_pt, row.center_pt]))
+    incoming['line'] = gpd.GeoSeries(lines, index=incoming.index ,crs= edges.crs)
+
+    #checking in there are more than one incoming lines arriving to the same point
+    #which would create several new lines
+    incoming['line_wkt'] = incoming.line.apply(lambda x : x.wkt)
+    count_s = incoming.groupby(['line_wkt'])['line_wkt'].count()
+    
+    #separating the incoming roads that come on their own to those that come in groups
+    filter_count_one  = pd.DataFrame(count_s[count_s == 1])
+    filter_count_many  = pd.DataFrame(count_s[count_s > 1])
+    incoming_ones = pd.merge(incoming, filter_count_one, left_on='line_wkt', right_index=True, how= 'inner')
+    incoming_many = pd.merge(incoming, filter_count_many, left_on='line_wkt', right_index=True, how= 'inner')
+
+    incoming_many_reduced = _coins_filtering_many_incoming(incoming_many, angle_threshold=angle_threshold)
+    incoming_all = gpd.GeoDataFrame(pd.concat([ incoming_ones, incoming_many_reduced]), crs = edges.crs)
+    
+    return incoming_all
+
+def _ext_lines_to_center(edges, incoming_all, rab_plus):
+    """Extends the Linestrings geometrie to the centerpoint defined by _rabs_center_points.
+    Also deleted the lines that originally defined the roundabout.
+    
+    Returns
+    -------
+    GeoDataFrame
+        GeoDataFrame of with updated geometry
+    """
+    ## this is causing a warning too for Shapely 2.0 --> Convert the '.coords' to a numpy array
+    incoming_all['geometry'] = incoming_all.apply(lambda row: linemerge([row.geometry, row.line]), axis =1)
+
+    # deleting the original round about edges
+    idx_out = edges.sjoin(rab_plus, predicate= 'covered_by', how='inner').index
```

Comment:
You have them selected above, you know their ID's. Try to reuse that and drop them based on the index instead of doing another spatial join. Those are expensive.